### PR TITLE
Factor out storage creation from SHM scripts

### LINKED ISSUE #1366: Factor out the creation of the storage account from the deployment script(s)
Factor out the creation of the storage account from the deployment script(s) into a separate script which can be run without having to deploy a full SRE (this can be run as part of deployment or separately)

### PR DESCRIPTION
**Pull request by craddm**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).
- [ ] You have marked this pull request as a **draft** and added `'[WIP]'` to the title if needed (if you're not yet ready to merge).
- [x] You have formatted your code using appropriate automated tools (for example `./tests/AutoFormat_Powershell.ps1 -TargetPath <path to file or directory>` for Powershell).

### :arrow_heading_up: Summary

Moves the SHM storage account creation step into its own script, adds it as a step in the `Deploy_SHM.ps1` script

### :closed_umbrella: Related issues

Closes #1366 

### :microscope: Tests

Tested SHM deployment and storage account creation with new scripts.

### CODE REVIEW

**Review by JimMadge — CHANGES_REQUESTED** [review]
Looks good, one comment about help text.

**Comment by craddm** [comment]
> Looks good, one comment about help text.

I can't see any comments or changes requested

**Comment by JimMadge** [comment]
🤦

File: deployment/safe_haven_management_environment/setup/Setup_SHM_Storage_Accounts.ps1 [hunk 1]
```diff
@@ -0,0 +1,40 @@
+param(
+    [Parameter(Mandatory = $true, HelpMessage = "Enter SHM ID (e.g. use 'testa' for Turing Development Safe Haven A)")]
```
Comment (JimMadge) [inline deployment/safe_haven_management_environment/setup/Setup_SHM_Storage_Accounts.ps1:2]: I think "testa" and "Turing Development Safe Haven A" won't make sense to anyone apart from us (and it sort of implies that "testa" is translated to the full name somehow).

How do we describe SHM ID in other help text?

**Review by JimMadge — CHANGES_REQUESTED** [review]

File: deployment/safe_haven_management_environment/setup/Setup_SHM_Storage_Accounts.ps1 (hunk 1 above)
Comment (craddm) [reply deployment/safe_haven_management_environment/setup/Setup_SHM_Storage_Accounts.ps1:2]: I see what you mean. The same help text is used everywhere, so will do a separate issue and change this in a separate PR

**Review by JimMadge — APPROVED** [review]

**Pull request merged**